src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -1207,12 +1279,41 @@ public String getWriteableName() {
     }
 
     /**
-     * Gets the field name that this query is searching against.
-     *
-     * @return The field name used in the Neural query
+     * Copy this QueryBuilder for two phase rescorer. This function will be invoked by the search processor
+     * NeuralSparseTwoPhaseProcessor which happens before the rewrite phase.
+     * @param pruneRatio the parameter of the NeuralSparseTwoPhaseProcessor, control the ratio of splitting the queryTokens to two phase.
+     * @param pruneType the parameter of the NeuralSparseTwoPhaseProcessor, control how to split the queryTokens to two phase.
+     * @return A copy NeuralQueryBuilder for twoPhase, it will be added to the rescorer.
      */
     @Override
-    public String fieldName() {
-        return this.fieldName;
+    public NeuralQueryBuilder getCopyQueryBuilderForTwoPhase(float pruneRatio, PruneType pruneType) {
+        this.neuralSparseQueryTwoPhaseInfo = new NeuralSparseQueryTwoPhaseInfo(
+            NeuralSparseQueryTwoPhaseInfo.TwoPhaseStatus.PHASE_ONE,
+            pruneRatio,
+            pruneType
+        );
+
+        // only copy the fields used for sparse embedding and available before the rewrite
+        final NeuralQueryBuilder copy = new NeuralQueryBuilder().queryName(queryName)
+            .fieldName(fieldName)
+            .queryText(queryText)
+            .modelId(modelId)
+            .searchAnalyzer(searchAnalyzer)
+            .neuralSparseQueryTwoPhaseInfo(
+                new NeuralSparseQueryTwoPhaseInfo(NeuralSparseQueryTwoPhaseInfo.TwoPhaseStatus.PHASE_TWO, pruneRatio, pruneType)
+            );
+
+        // If raw tokens are provided directly in the query, split them without additional processing
+        if (Objects.nonNull(this.queryTokensMapSupplier)) {
+            Map<String, Float> tokens = queryTokensMapSupplier.get();
+            // Splitting tokens based on a threshold value: tokens greater than the threshold are stored in v1,
+            // while those less than or equal to the threshold are stored in v2.
+            Tuple<Map<String, Float>, Map<String, Float>> splitTokens = PruneUtils.splitSparseVector(pruneType, pruneRatio, tokens);
+            this.queryTokensMapSupplier(splitTokens::v1);
```

Comment:
nit. Not ideal because the method name indicate just getting copy of query builder but this actually modify this.queryTokensMapSupplier.